src/net.cpp

```diff
@@ -2837,13 +2835,12 @@ void CConnman::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
         if (message_size) {
             pnode->vSendMsg.push_back(std::move(msg.data));
         }
-
-        // If write queue empty, attempt "optimistic write"
-        if (optimistic_send) {
-            bytes_sent = SocketSendData(pnode);
-        }
     }
-    if (bytes_sent) RecordBytesSent(bytes_sent);
+    // If write queue was empty, attempt "optimistic write"
+    if (optimistic_send) {
+        const size_t bytes_sent{SocketSendData(pnode)};
```

Comment:
Why is this change needed? Previously we'd optimistically write only one message, now we might write more than one message when other messages have been pushed by other threads.